src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1772,7 +1772,7 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, SyntaxToken tar
 
             var enclosingSymbol = semanticModel.GetEnclosingSymbol(targetToken.SpanStart, cancellationToken);
 
-            while (enclosingSymbol is IMethodSymbol method && method.MethodKind == MethodKind.LocalFunction)
+            while (enclosingSymbol is IMethodSymbol method && (method.MethodKind == MethodKind.LocalFunction || method.MethodKind == MethodKind.AnonymousFunction))
             {
                 // It is allowed to reference the instance (`this`) within a local function, as long as the containing method allows it
```

Comment:
very tiny nit: consider updating comment to say "within a local or anonymous function"

Reply:
wouldn't is just be simpler to do ```while (enclosingSymbol is IMethodSymbol && enclosingSymbol.ContainingSymbol is IMethodSymbol) { ... }```?

Reply:
Effective, walk up to the outermost IMethodSymbol.

Reply:
Hmm... maybe, though you have to think a little more about all the cases where this could occur, so maybe a little harder to reason about

Reply:
For example. what if you're in a lambda function but not in a method, let's say a field initializer? I feel like thsi might give differnt results, since it will just stay at the lambda as the outermost method symbol, even though we know it's a lambda and we don't really want to consider it, we should still walk up.

Reply:
I think the knowledge that we have a local function or lambda is important since that means we have to walk up, no matter whether we are in a method or not.